### PR TITLE
lsp: Add snippets for new window and dialog

### PR DESCRIPTION
**Pull request by hunger**
Makes starting fresh a tiny bit simpler IMHO...

Ideally we would define file templates, but that is not the VSCode way:-( So offer soem simple snippets instead.

### CODE REVIEW

File: tools/lsp/language/completion.rs [hunk 1]
```diff
@@ -1204,7 +1206,6 @@ mod tests {
             "component Bar in🔺 Window {}",
         ];
         for source in sources {
-            eprintln!("Test for inherits in {source:?}");
```
Comment (ogoffart) [inline tools/lsp/language/completion.rs:1207]: why remove this?
If the test fails, this tells what iteration of the loop caused the failure.
If the test doesn't fail, the output is being captured and not shown.

File: tools/lsp/language/completion.rs [hunk 2]
```diff
@@ -341,6 +341,8 @@ pub(crate) fn completion_at(
             ("export component", "export component ${1:ExportedComponent} {\n    $0\n}"),
             ("export struct", "export struct ${1:Name} {\n    $0\n}"),
             ("export global", "export global ${1:Name} {\n    $0\n}"),
+            ("new window", "export component ${1:MyMainWindow} inherits Window {\n    $0\n}"),
+            ("new dialog", "export component ${1:MyDialog} inherits Dialog {\n    $0\n}"),
```
Comment (ogoffart) [inline tools/lsp/language/completion.rs:345]: We should not add this if the editor don't support snippet.  For the others string, the first string will be completed if the editor don't support snippets, and that's ok. But here, "new window" makes no sense to complete literally)

**Review by ogoffart — COMMENTED** [review]
Personally, I'm not a fan of this.
creating a new window is something you basically only do once in a project. And you usually do start with a template that already has it. 
Also how will people know that they can type "new window" and that doesn't mean "new window" and that "new" is not anything special in Slint.
Since at this point we only support one window, this can be misleading. 
But I'd not use an auto-completion for this, but rather a code action.

**Comment by tronical** [comment]
> creating a new window is something you basically only do once in a project. And you usually do start with a template that already has it.

I'm with Olivier on this. I think in a programming environment like VS Code, we should always start with our Rust, C++, or Node.js project templates.

**Pull request closed without merging**